gstreamer/src/caps.rs

```diff
@@ -397,6 +397,67 @@ impl Builder {
     }
 }
 
+#[cfg(feature = "ser_de")]
+pub(crate) mod serde {
+    use serde::de::{Deserialize, Deserializer, SeqAccess, Visitor};
+    use serde::ser::{Serialize, Serializer, SerializeSeq};
+
+    use std::fmt;
+
+    use Caps;
+    use CapsRef;
+    use Structure;
+
+    impl<'a> Serialize for CapsRef {
+        fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+            let iter = self.iter();
+            let size = iter.size_hint().0;
```

Comment:
Strictly speaking, next to each structure the caps also have an optional `GstCapsFeature`. That API is not part of the bindings yet, so just add a FIXME comment about that somewhere here

Reply:
Would that lead to something like:
```
(
    structures: [
    ],
    features: Some([
    ]),
)
```

If so, we might want to include both fields right now and keep `features: None` until `GstCapsFeautre` is part of the binding. That would be future proof. Alternatively, we could just remove the `Option` part and just keep the array, which can be empty.

Reply:
It would be more like
```
structure: [
    (structure1, feature1),
    (structure2, feature2),
],
```